cloudfront/wellcomelibrary.org/edge-lambda/src/safeGet.ts

```diff
@@ -0,0 +1,27 @@
+import axios, { AxiosRequestConfig, AxiosResponse } from 'axios';
+
+export async function safeGet<T>(
+  url: string,
+  config?: AxiosRequestConfig
+): Promise<Error | T> {
+  const apiResult: AxiosResponse<any> | Error = await axios
+    .get(url, config)
+    .catch((error) => {
+      // Log underlying error here, we don't want to expose it to callers.
+      console.error(error);
+
+      if (error.response) {
+        return Error(`Got ${error.response.status} from ${url}`);
+      } else if (error.request) {
+        return Error(`No response from ${url}`);
+      }
+
+      return Error(`Unknown error from ${url}: ${error}`);
+    });
+
+  if (apiResult instanceof Error) {
+    return apiResult;
+  }
+
+  return apiResult.data;
```

Comment:
Is this going to bin the response code? Do we care? e.g. a 302 redirect

Reply:
Yes, it will, but I think in this function we are only interested in responses that return data and the status isn't important. I believe that axios will be default [follow redirects](https://github.com/axios/axios/blob/8b0f373df0574b7cb3c6b531b4092cd670dac6e3/test/unit/adapters/http.js#L98).